### PR TITLE
Merging oam-runtime into vela-core

### PR DESCRIPTION
**Pull request by wonderflow**
- [x] 1. merge apis,pkg libs
- [x] 2. change imports headers
- [x] 3. merge tests
- [x] 4. merge main.go, including all controllers
- [x] 5. merge scripts, makefiles, charts
- [x] 6. CI checks

---
 docs, design, examples will not merge

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -181,6 +181,10 @@ issues:
       linters:
         - errcheck
 
+    - text: "don't use an underscore"
```
Comment (hongchaodeng) [inline .golangci.yml:184]: What's this for?

File: apis/generate.go [hunk 2]
```diff
@@ -3,6 +3,22 @@
 // See the below link for details on what is happening here.
 // https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
 
+// NOTE(@negz): See the below link for details on what is happening here.
```
Comment (hongchaodeng) [inline apis/generate.go:6]: Remove this line?
It's duplicate to above line.

**Review by hongchaodeng — APPROVED** [review]
lgtm

File: .golangci.yml (hunk 1 above)
Comment (wonderflow) [reply .golangci.yml:184]: package core.oam.dev can only used by `core_oam_dev` while golint complain this message

File: apis/generate.go (hunk 2 above)
Comment (wonderflow) [reply apis/generate.go:6]: sure, thanks

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -2,26 +2,93 @@ name: Go
...
+
+      - name: Cache Go Dependencies
+        uses: actions/cache@v2
+        with:
+          path: .work/pkg
+          key: ${{ runner.os }}-pkg-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-pkg-
+
+      - name: Check Diff
+        run: make check-diff
```
Comment (hongchaodeng) [inline .github/workflows/go.yml:70]: This will run cue-fmt, which depends on the cue tool.
Maybe move cue-fmt into separate make command?

Comment (wonderflow) [reply .github/workflows/go.yml:70]: cue-fmt also cause file changes, I hope we can check all of them, let me solve the cue tool dependency

Comment (wonderflow) [reply .github/workflows/go.yml:70]: added

**Pull request merged**